lib/cn/cn_tree.c

```diff
@@ -1812,15 +1683,14 @@ cn_tree_cursor_create(struct cn_cursor *cur, struct cn_tree *tree)
      * The logic for descending the tree is similar to the logic
      * cn_tree_lookup().
      */
-
-    node = tree->ct_root;
-
     rmlock_rlock(&tree->ct_lock, &lock);
     cur->dgen = cn_get_ingest_dgen(cur->cn);
+    node = tree->ct_root;
+
     while (node) {
 
         /* recover least dgen of parent when entering a node */
-        u32 level = node->tn_loc.node_level;
+        // is the above "least-dgen" comment stale???
```

Comment:
Maybe throw an HSE_REVISIT in there

Reply:
Will do unless @gsramdasi has some guidance...

Reply:
@gsramdasi any comment here?  Does github send a note if you reference someone in a comment???